core/model-vocabulary/src/main/java/org/eclipse/rdf4j/model/vocabulary/CONFIG.java

```diff
@@ -0,0 +1,358 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eclipse RDF4J contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+package org.eclipse.rdf4j.model.vocabulary;
+
+import org.eclipse.rdf4j.model.IRI;
+
+/**
+ * Shared vocabulary for configuration of RDF4J components: Repositories, SAILs, and so on.
+ *
+ * @author Jeen Broekstra
+ *
+ * @since 4.3.0
+ */
+public class CONFIG {
+
+	/**
+	 * The RDF4J config namespace (<var>tag:rdf4j.org,2023:config/</var>).
+	 *
+	 * @see <a href="https://tools.ietf.org/html/rfc4151">the 'tag' URI Scheme (RFC 4151)</a>
+	 */
+	public static final String NAMESPACE = "tag:rdf4j.org,2023:config/";
+
+	/**
+	 * The recommended prefix for the RDF4J config namespace: "config"
+	 */
+	public static final String PREFIX = "config";
+
+	/**
+	 * Setting for linking a delegate config to a wrapper in a SAIL or Repository config stack.
+	 *
+	 * <var>tag:rdf4j.org,2023:config/delegate</var>
+	 */
+	public final static IRI delegate = Vocabularies.createIRI(NAMESPACE, "delegate");
+
+	/**
+	 * Repository config
+	 */
+	public static final class Rep {
```

Comment:
Still on the fence if having these as nested class is an improvement. May revisit to separate these out again.